src/Scenes/Opening.cpp

```diff
@@ -163,13 +174,72 @@ void Opening::initTraffic()
 
 std::uint16_t Opening::poll(const sf::Event& e)
 {
+
     if (e.type == sf::Event::KeyPressed)
     {
+        updateMessage();
+
         switch (e.key.code)
         {
             default:
             break;
 
+            //
+            // Action. Perform an action. E.g. pick up an item.
+            //
+            case sf::Keyboard::A:
+            {
+                //
+                // Check if we are on an item.
+                // If yes, remove it from the map's _items and add it to the 
+                // player's inventory.
+                //
+                for(auto it = _items.begin(); it != _items.end(); it++)
+                {
+                    ItemPtr item = *it;
+
+                    if( item->getGlobalBounds().intersects(
+                                                _player->getGlobalBounds()) &&
+                        item->isObtainable() )
+                    {
+
+                        _player->addItem(item->getID());
+                        _missionText.setText("Picked up " + item->getName());
+                        _items.erase(it);
+
+                        break;
+                    }
+                }
+            }
+            break;
+
+            //
+            // Inventory. Display inventory.
+            //
+            case sf::Keyboard::I:
+            {
+                std::map<std::string, std::int32_t> inv =
+                                                    _player->getInventory();
+                std::vector<std::string> keys;
+
+                for(auto it = inv.begin(); it != inv.end(); it++)
```

Comment:
Just an FYI, not a change request, there's some cool syntatic sugar in modern C++ that makes this kind of thing prettier. For example:

```
for (const auto& it : _inv)
{
    keys.push_back(it->first);
}
```